### PR TITLE
ci: set up pre-commit

### LINKED ISSUE #466: ci/staticcheck: Add shell scripts linters to CI
> * Lint with `shellcheck` and `hadolint`

We should definitely add these checks both in parca and parca-agent!

_Originally posted by @kakkoyun in https://github.com/parca-dev/parca-agent/issues/465#issuecomment-1157271684_

cc @maxbrunet

### PR DESCRIPTION
**Pull request by maxbrunet**
Similar to parca-dev/parca#1495

Closes #466

### CODE REVIEW

File: e2e/local-e2e.sh [hunk 1]
```diff
@@ -1,5 +1,18 @@
 #!/bin/bash
```
Comment (kakkoyun) [inline e2e/local-e2e.sh:1]: cc @Sylfrena, you might want to check these changes. FYI They look good to me.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - id: insert-license
+        name: Insert license in comments (Shell)
+        args:
+          - --license-filepath=scripts/LICENSE.txt
+          - --comment-style=#
+          - --fuzzy-match-generates-todo
+        types: [shell]
+        exclude: ^internal/(?:go|pprof)/
+  - repo: https://github.com/golangci/golangci-lint
+    rev: v1.48.0
```
Comment (kakkoyun) [inline .pre-commit-config.yaml:29]: Is there a way to use renovate to keep track of these versions as well? We will have version drifts otherwise.

**Review by kakkoyun — COMMENTED** [review]
Overall LGTM.

I have a question, and we need to rebase.

File: e2e/local-e2e.sh (hunk 1 above)
Comment (Sylfrena) [reply e2e/local-e2e.sh:1]: LGTM as well, definitely learnt some nifty scripting/ci stuff from the PR. :D

**Review by Sylfrena — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (maxbrunet) [reply .pre-commit-config.yaml:29]: Yes! Remember parca-dev/.github#5?

Comment (kakkoyun) [reply .pre-commit-config.yaml:29]: Ah yeah, true! 👍

**Pull request merged**